### PR TITLE
Separate search from landing, handle failed requests

### LINKED ISSUE #26: Separate search from landing
It is not very desirable to force the search page to be dynamically loaded within the landing page.

The primary goal would be to avoid hitting the user with a megabyte-sized JSON hammer on landing. Unfortunately, it would be impossible to do so without renouncing the Cask counter, as the GitHub API seems not to paginate content responses.

### LINKED ISSUE #31: Hitting Github API rate limit causes TypeError: Cannot read property 'filter' of undefined
When using the search feature on the website, I get the following exception in the console from [cask.js:112](https://github.com/caskroom/caskroom.github.io/blob/8cb4866bb44b2f22a99f22edbc2a7ca93e12ca32/js/cask.js#L112).

```
Uncaught TypeError: Cannot read property 'filter' of undefined
```

Looks like I'm also hitting the rate limit for the Github api which prevents the formulas from being brought in which results in caskList being undefined.

### PR DESCRIPTION
**Pull request by ndr-qef**
Monolithic update.

If GitHub API fails to respond,
- the splash page will do nothing;
- the search page will display a message, either generic or specific to rate-limitation.

Closes #26, fixes #31.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -104,18 +104,6 @@ <h2 class="beta ale">Get Cask</h2>
                 <a href="/search" id="search-button" class="textual">search</a>
```
Comment (radeksimko) [inline index.html]: The new filename is `search.html`, so this imo needs changing, otherwise it will end up in `404`.

**Comment by radeksimko** [comment]
Generally I like it, but it would be nice to use the header Github gives us in each response, [`X-RateLimit-Reset`](https://developer.github.com/changes/2013-07-02-rate-limit-reset/) in particular, so the user can see something useful, like _"try again in 21 minutes"_ or something similar.

File: index.html (hunk 1 above)
Comment (ndr-qef) [inline index.html]: It will not once it is online: GitHub handles the prettifying of routes automatically.

**Comment by ndr-qef** [comment]
Indeed, that would be the plan. I didn't quite have the time to test it, however, and preventing errors has a sufficiently high priority not to satisfy all concerns immediately.

Merging with memo to include the reset time.

File: index.html (hunk 1 above)
Comment (radeksimko) [inline index.html]: Ah, I see, although it doesn't work with local jekyll instance.

**Pull request merged**